README.md

```diff
@@ -279,15 +284,45 @@ Detects changed files that have not yet been committed. This does **not** includ
 
 Detects files that are not yet tracked by git (see `git status` manual). This does **not** include _uncommitted_ files. A new file is not _untracked_ anymore after it is added to the index.
 
+### gib.buildAll
+
+Builds all modules. Can be used to (temporarily) override the reduction of modules to build.
+
+Can be combined/useful with `gib.skipTestsForNotImpactedModules` and/or `gib.argsForNotImpactedModules`.
+
+### gib.buildDownstream
+
+By default, GIB not only builds modules that contain changes but also the modules that depend on those changed modules (donwstream modules of the changed modules).
+
+Setting this property to `false` excludes such downstream modules.
+
+GIB does _not_ evaluate Maven core `also-make-dependents` mode (`mvn -amd`).
+
+### gib.buildUpstreamMode
+
+In conjunction with `mvn -am` this property allows to control the selection of upstream modules (= modules that the modules GIB has determined to build _depend on_):
+
+- `changed` (default value): selects only upstream modules of the _directly changed_ modules
+- `impacted`: like `changed` but also selects upstream modules of modules _that depend on_ the directly changed modules (in other words: upstream modules of the downstream modules of the changed modules)
+
+`changed` is a subset of `impacted`.
+
+`impacted` may seem odd at first, but it does come in handy in certain scenarios, e.g. a Jenkins PR job that locally merges target branch into the PR branch before building.
+Here it might be required to freshly compile upstream modules of not directly changed modules to avoid compile errors or test failures which originate from the target branch.
+
+Both strategies can and usually should be combined with `gib.skipTestsForNotImpactedModules` and/or `gib.argsForNotImpactedModules`.
+
+Note: _Before_ 3.8, GIB did non have this property and was implicity applying the `impacted` strategy, see also #44.
+
 ### gib.skipTestsForNotImpactedModules
 
-In conjunction with `-am` or `gib.buildAll=true` this property disables the compilation/execution of tests for modules that have _not_ been changed by adding `maven.test.skip=true`. In case a not impacted module produces a test jar just the test _execution_ is disbled via `skipTests=true`.
+In conjunction with `mvn -am` or `gib.buildAll=true` this property disables the compilation/execution of tests for modules that have _not_ been changed by adding `maven.test.skip=true`. In case a not impacted module produces a test jar just the test _execution_ is disabled via `skipTests=true`.
```

Comment:
The property and so others still contains "Impacted". Why not rename it to skipTestsForNotDownstream to migrate to downstream/upstream fully?

Reply:
Yes, depricating first sounds good.

Reply:
I pushed the name adjustment + deprecation.